### PR TITLE
State of the schema sprint

### PR DESCRIPTION
**Pull request by effigies**
Mostly doing this to get an RTD pull request build going as we update.

### CODE REVIEW

File: src/schema/rules/dataset_metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    DatasetDOI: OPTIONAL
+    GeneratedBy: RECOMMENDED
+    GeneratedBy[].Name: REQUIRED
+    GeneratedBy[].Version: RECOMMENDED
+    GeneratedBy[].Description: OPTIONAL
+    GeneratedBy[].CodeURL: OPTIONAL
+    GeneratedBy[].Container: OPTIONAL
+    GeneratedBy[].Container{}.Type: RECOMMENDED
+    GeneratedBy[].Container{}.Tag: RECOMMENDED
+    GeneratedBy[].Container{}.URI: RECOMMENDED
```
Comment (Remi-Gau) [inline src/schema/rules/dataset_metadata.yaml:32]: Those dot, brackets and curly braces will MATLAB cry: the only special characters allowed in keys by matlab are `_` 

Related to 
https://github.com/bids-standard/bids-specification/pull/1082

Comment (effigies) [reply src/schema/rules/dataset_metadata.yaml:32]: We need some way to mark the requirement levels of sub objects.

Comment (Remi-Gau) [reply src/schema/rules/dataset_metadata.yaml:32]: See also: https://github.com/bids-standard/bids-specification/issues/841

Comment (tsalo) [reply src/schema/rules/dataset_metadata.yaml:32]: #841 is really just an artifact at this point. Once someone (probably me) writes a function to render subobjects in tables, we can get rid of the `Genetics.*` fields.

**Comment by effigies** [comment]
Three live :+1:s. Merging to avoid accumulating merge conflicts.

**Pull request merged**